### DISCUSSION TITLE
Global var not accessable in both functions

### DISCUSSION THREAD

**Discussion by ChrisVonFrosta** [arendst/Tasmota#17977] [Q&A]
i have a function.be as modul that ist imported from web.be.
web.de is only for triggering a `start()` from function.be. You could see the startingpoint in this [discussion](https://github.com/arendst/Tasmota/discussions/17963)

 the function `start()` calls the function `do()`
both in function.be

 `do()` edit the global var  `state`

 `start()` call this `do()` and should react to  `state` .
depends on the  `state` value coming from `do()`,  `start()` calls his self again with 5000ms delay. 
but the  `state` is allways 0  in  `start()`.
even if do() set it to 1 or -1.

 ```
#function.be
var function_module = module("bewasserung_module")  #import from web.be

var state=0

...

def do()
...
  state=read_and_proceed_sensor()
  print(state)
...
end

function_module .start = def ()   #initial call from web.be
...
    if state == 0
        do()    # state return from sensor, set state depending from sensor value
        printf("still not good)
        tasmota.timeout(5000,start)`
    elif
        print("ready to do next step things")
    else
        print("finish")
     end
...
end

start()

return function_module
````
so i end in a endless periode with `state==0`.

i could work arround by returning  `state` value from `do()` instead using global var. but i interessted why this happen or could work with globak vars.

Thx a lot for sharing time and acknowledgement!

**Comment by sfromis** [comment]
It is not clear what code you have with syntax errors (including undefined `tasmota.timeout`), and unclear cuts in what you posted. Not repeatable to check up on what happens.

With `do` being a reserved word in Berry, you should probably not use that as a function name.
More generally, avoiding a technique like passing data via global vars is seen as good programming practice.

**Reply by ChrisVonFrosta** [reply]
sorry for that. it was late here.
i will send an reproducable example next days ( i have to go).

but it seems that "normal"functions and module functions  can't share vars at teh same time.
using a var in just functons works.
using this var in both makes it zero in the module function...

**Comment by ChrisVonFrosta** [comment]
i worked around with returning values and local variables
